### ISSUE TITLE
Code of your TPAMI version -- CoVR2

### ISSUE THREAD

**Issue by zinuoli** [lucas-ventura/CoVR#21]
Hi, thanks for this great work.

Do you have any plan to release the code of your TPAMI version? I notice the performance has been increased a lot.

Maybe I miss some parts of your repo about the CoVR2 : ) If I do, please forgive me.

**Comment by zinuoli** [comment]
Thank you so much for such quick reply, and the push of vew code! I like this work very much : )

**Comment by lucas-ventura** [comment]
I’m not sure I fully understand your question, but here’s how you can switch to CoVR2 and get started:

```bash
git pull
git switch covr2
git pull
```

To train or test with CoVR2, you'll first need to extract BLIP-2 embeddings. You can find more details in the [usage section of the CoVR2 branch](https://github.com/lucas-ventura/CoVR/tree/covr2?tab=readme-ov-file#usage-computer).

If you want to train with CoVR2, you can use the following command:

```bash
python train.py model=blip2-coco model/ckpt=blip2-l-coco
```

There are a few predefined CoVR2 configurations in the [`experiments2`](https://github.com/lucas-ventura/CoVR/tree/covr2/configs/experiment2) folder. For instance, to train with WebVid-CoVR dataset, you can run:

```bash
python train.py experiment2=covr
```

To evaluate CoVR2 models, you can download them with:

```bash
bash tools/scripts/download_pretrained_models.sh
```

If you want to test any specific models, let me know, and I can provide more detailed instructions.

**Comment by zinuoli** [comment]
That's exactly what I want to know, thank you for that.

**Comment by zinuoli** [comment]
Hi, sorry for re-open again. The new environment stated in requirements.txt for CoVR2 seems incompatible, could you please check it by creating a new environment from scratch? Looks like the problem happens between numby and opencv-python-headless.

**Comment by zinuoli** [comment]
Also, the `test.py` is a bit confusing, it seems to be using BLIP by default instead of BLIP2. Could you please provide me with an example to use BLIP2 to test on `webvid-covr`?

**Comment by lucas-ventura** [comment]
Sorry about that! You’re right, the ``requirements.txt`` was not working. I’ve just updated it in my latest push: c9423a0e122d2bd1676587d6d2498540c13a6a42.

Please try setting up the environment again and let me know if it resolves the problem.

Thank you for catching that!

**Comment by zinuoli** [comment]
Thank you so much for replying to me, I will try tomorrow : ) Before this I solved this problem by forcing pip to install old verion `numpy`, problem solved and evaluation ran successfully, but numpy becomes incompatible with other packages again.

Hope this new env will have better compatibility 😄 Other parts of the codes seem to run well from my side.

**Comment by zinuoli** [comment]
Hi, sorry for disturbace again.

Just wondering if you provide BLIP2 config that is exclusively for `WebVid-CoVR`? I notice that you provide 3 options `model/ckpt=blip2-l-coco``model/ckpt=blip2-l-coco_coir``model/ckpt=blip2-l-coco_coir+covr`, but none of them is only for CoVR.

I have created a `yaml` and managed to test only on `WebVid-CoVR` using the `ckpt4.pth`, because you provide download way for the ckpt. But for the `model/ckpt=blip2-l-coco_coir+covr`, it is using `ckpt2.pth` not `ckpt4.pth`.

The performance looks good but I am doing not officially, so just wonder if I missed something or you didn't provide such config file?

Thanks a lot for this work, so detailed and so nice.

**Comment by yuqianzzz** [comment]
Hi, thanks for this amazing work:)) would be grateful if you can answer this little question:
experiment2=covr is setting batch size as 256 and lr as 2e-5 which is not the description what paper offered. Is there any mistake? (When i use same hardware settings to train on webvid-covr, the cuda memory is not enough.)

**Comment by lucas-ventura** [comment]
Hi,

It is a mistake. I believe the correct value for the learning rate should be 2e-5, but I recall that both 1e-5 and 2e-5 gave similar results. I'll re-run the models in the coming days to confirm and will let you know.

Regarding the batch size, the batch size specified in the config is per device, so it's actually 256 * 4 = 1024 in total across all devices.

Thanks for catching that! I'll update the numbers once I have the results.
